backend/src/core/features/game-identifier/game.identifier.spec.js

```diff
@@ -1,715 +1,499 @@
 import { GameIdentifier } from "./game.identifier.js";
-import { animaddicts2gameHtmlDetailsPage } from "../../../../assets/steam-details-pages/animaddicts.2.game.html.details.page.js";
-import { glitchhikersSoundtrackHtmlDetailsPage } from "../../../../assets/steam-details-pages/glitchhikers.soundtrack.html.details.page.js";
-import { eldenRingHttpDetailsSteamcharts } from "../../../../assets/steamcharts-details-pages/elden.ring.multiple.histories.html.details.page.js";
-import { SteamApp } from "../../models/steam.app.js";
-import {
-  discoverGamesFromSteamWeb,
-  identifyGames,
-  recordAttemptsViaSteamDb,
-  updateMissingReleaseDates,
-  updateTypeSideEffectFree,
-} from "../../services/game.service.js";
 import { HistoryCheck } from "../../models/history.check.js";
 import { createLoggerMock } from "../../../common/logger.mock.js";
 import { counterStrikeHtmlDetailsSteamDb } from "../../../../assets/steamdb-details-pages/counter.strike.html.details.page.js";
 import { riskOfRainHtmlDetailsSteamDb } from "../../../../assets/steamdb-details-pages/risk.of.rain.html.details.page.js";
 import { getXGamesWithoutDetails } from "../../models/game.mocks.js";
 import { createConfigMock } from "../../../common/config.loader.mock.js";
 import { getXSampleSteamApps } from "../../models/steam.app.mocks.js";
-import { createHtmlDetailsPages } from "../../../../assets/html.details.pages.mock.js";
+import { ValidDataSources } from "../../models/valid.data.sources.js";
+import { gta5ageRestrictedHtmlDetailsPage } from "../../../../assets/steam-details-pages/gta.5.age.restricted.html.details.page.js";
+import { theSims4dlcHtmlDetailsPage } from "../../../../assets/steam-details-pages/the.sims.4.dlc.html.details.page.js";
+import { getParsedHtmlPages } from "../../../../assets/html.details.pages.mock.js";
+import { mortalDarknessGameHtmlDetailsPage } from "../../../../assets/steam-details-pages/mortal.darkness.game.html.details.page.js";
+import { SteamAppsAggregate } from "../../models/steam.apps.aggregate.js";
+import { GamesAggregate } from "../../models/games.aggregate.js";
+import { parseHTML } from "linkedom";
 
 describe("game.identifier.js", function () {
-  describe(".tryViaSteamWeb", function () {
-    describe("gets zero steamApps from the database and stops. So,", function () {
-      beforeAll(async function () {
-        this.steamClientMock = createSteamMock([undefined]);
-
-        this.steamAppsRepository = createSteamAppsRepositoryMock([], undefined);
-        this.gamesRepository = createGamesRepositoryMock();
-        this.historyChecksRepository = createHistoryChecksRepositoryMock();
+  describe(".checkIfGameViaSource.", function () {
```

Comment:
You could heavily simplify the tests. No need to check for the calling or not calling of all methods of the mocks, it would be enough to check for the flow of the logic, e.g. when isEmpty is true => just check that nothing is persisted. And if isEmpty is false check only for the expected result to be persisted. Do you know what I mean?

Reply:
Will skip this for now since API will change, opened ticket to do it after: https://github.com/lukatarman/steam-game-stats/issues/212